### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by seadowg**
Some updates to our docs based (and one to the default gradle config). I've left comments inline on hopefully justifying any changes I've made but these are all very open for discussion.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -55,22 +57,19 @@ To contribute code to ODK Collect, you will need to open a [pull request](https:
 
 1. Keep your pull request focused on one narrow goal. This could mean addressing an issue with multiple, smaller pull requests. Small pull requests are easier to review and less likely to introduce bugs. If you would like to make stylistic changes to the code, create a separate pull request.
 
-1. Run `./gradlew checkCode` and fix any errors.
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:58]: This is part of the PR template tasks and doesn't feel to me like it needs to be highlighted here

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -55,22 +57,19 @@ To contribute code to ODK Collect, you will need to open a [pull request](https:
...
-
 1. Write clear code. Use descriptive names and create meaningful abstractions (methods, classes).
 
 1. Document your reasoning. Your commit messages should make it clear why each change has been made.
 
 1. Point out decisions you made and what alternatives you considered. If you're unsure about a particular approach, ask a question to make your own thinking clear and help the reviewer identify controversial parts of the proposed solution. For example: "here I returned a result object to represent the status after the transaction. I also considered throwing an exception in case of error but I didn't like that it made it unclear where an error happened. Which do you prefer and why?" This is particularly important for [work in progress pull requests](#work-in-progress-pull-requests).
 
-1. If your pull request makes user-facing changes, we likely need to update documentation. [File an issue on the docs repo](https://github.com/getodk/docs/issues/new) describing the changes.
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:66]: Again this is in the PR template tasks

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -55,22 +57,19 @@ To contribute code to ODK Collect, you will need to open a [pull request](https:
...
-
 1. Follow the guidelines below.
 
 ## The review process
-Bug fixes, pull requests corresponding to issues with a clearly stated goal and pull requests with clear tests and/or process for manual verification are given priority. Pull requests that are unclear or controversial may be tagged as `needs discussion` and/or may take longer to review.
 
-We try to have at least two people review every pull request and we encourage everyone to participate in the review process to get familiar with the code base and help ensure higher quality. Reviewers should ask themselves some or all of the following questions:
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:73]: We don't require two people as far as I've seen. Maybe we should add that it requires a review from a committer? I guess this is kind of implicit given a committer is required to merge.

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -80,19 +79,19 @@ We try to have at least two people review every pull request and we encourage ev
 - Is the code easy to understand and to maintain?
 - Is there sufficient detail to inform any changes to documentation?
 
-When a pull request is first created, @getodk-bot tags it as `needs review` to indicate that code review is needed. Community members review the code and leave their comments, verifying that the changes included are relevant and properly address the issue. A maintainer does a thorough code review and when satisfied with the code, tags the pull request as `needs testing` to indicate the need for a manual [black-box testing](https://en.wikipedia.org/wiki/Black-box_testing) pass. A pull request may go back and forth between `needs testing` and `needs review` until the behavior is thoroughly verified. Once the behavior has been thoroughly verified, the pull request is tagged as `behavior verified`. A maintainer then merges the changes. Pull requests that need more complete reviews including review of approach and/or appropriateness are tagged with `reviews wanted`. Any community member is encouraged to participate in the review process!
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:83]: We don't use the `needs review` label anymore. Guessing Github has got better at flagging this? I definitely don't feel like it's something we need to be doing - any PR that doesn't have a review needs reviewed.

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -80,19 +79,19 @@ We try to have at least two people review every pull request and we encourage ev
...
+Community members review the code and leave their comments, verifying that the changes included are relevant and properly address the issue. A maintainer does a thorough code review and when satisfied with the code, accepts and may label the pull request as `needs testing` to indicate the need for a manual [black-box testing](https://en.wikipedia.org/wiki/Black-box_testing) pass. A pull request may have `needs testing` removed and added again several times until the behavior is thoroughly verified. Once the behavior has been verified, the pull request is labelled as `behavior verified`.
 
-Small fixes that target very particular bugs may occasionally be merged without a second review.
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:85]: Got rid of this seeing as I removed the need for a second review.

File: docs/CONTRIBUTING.md [hunk 6]
```diff
@@ -80,19 +79,19 @@ We try to have at least two people review every pull request and we encourage ev
...
+## Merging pull requests
 
-## Work in progress pull requests
+Pull requests can be merged by committers once they are accepted or once they have been accepted and have been labelled as `behavior verified` unless:
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:86]: Added this to make it explicit that CI has to be green for us to merge. Also thought it was good to specify that only committers can merge and get a mention of Code Freeze in here (currently only in `README.md`).

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -80,19 +79,19 @@ We try to have at least two people review every pull request and we encourage ev
...
 
-Work in progress (WIP) pull requests are useful to illustrate a proposed direction and get early feedback before committing to a solution direction. These should ideally ideally be created as [Draft Pull Requests](https://help.github.com/en/articles/about-pull-requests#draft-pull-requests). 
+* There is a currently a "code freeze" due to an [ongoing release](/README.md#release-cycle)
+* [CI](https://app.circleci.com/pipelines/github/getodk/collect) is currently failing
+
+## Work in progress pull requests
 
-If a pull request is already a non-draft (there is currently no way to move it back to a draft) the PR:
```
Comment (seadowg) [inline docs/CONTRIBUTING.md:91]: Github fixed this 🎉

File: docs/STATE.md [hunk 8]
```diff
@@ -16,17 +16,18 @@ and update this document as the code evolves.
 ## Where we are now
 
 * App written in a mixture of Java and Kotlin
-* App has mixture of unit tests (JUnit), Robolectric tests (Junit + Robolectric) and Espresso tests but coverage is far from complete
+* App has mixture of unit tests, ["Local tests" and "Instrumented tests"](https://developer.android.com/training/testing/fundamentals#instrumented-vs-local) but coverage is far from complete
```
Comment (seadowg) [inline docs/STATE.md:19]: Android docs now have clearer language on this split so wanted to reflect that here.

File: docs/STATE.md [hunk 9]
```diff
@@ -37,9 +38,10 @@ and update this document as the code evolves.
...
 
 * General effort to increase test coverage and quality while working on anything and pushing more for tests in PR review
-* Slowly moving responsibilities out of FormEntryActivity
+* Slowly moving responsibilities out of `FormEntryActivity`
 * Writing pretty much all new code in Kotlin
 * Writing new code using a [multi-module approach](CODE-GUIDELINES.md#gradle-sub-modules) (feature modules, mini frameworks etc) and breaking old code out into modules when opportunities come up
 * Trying to remove technical debt flagged with `@Deprecated`
 * Replacing Rx (and other async work such as `AsyncTask`) with `LiveData` + `Scheduler` abstraction
 * Gradually removing use of `CursorLoader` (all remaining uses are in `CursorLoaderFactory`)
+* Using AndroidX Test in new local tests and migrating other local tests as we touch them (from classic Robolectric)
```
Comment (seadowg) [inline docs/STATE.md:47]: We haven't talked about this, but it's definitely the direction we're going in

File: gradle.properties [hunk 10]
```diff
@@ -2,4 +2,5 @@
 android.useAndroidX=true
 android.enableJetifier=true
 org.gradle.jvmargs=-Xmx1g
+org.gradle.parallel=true
```
Comment (seadowg) [inline gradle.properties:5]: Realized people might not be benefiting from this. This is paying more and more dividends as we commit more to multimodule as each one can have tasks run on them in parallel.

**Review by lognaturel — APPROVED** [review]
Thank you!!

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**